pandas/tests/indexes/test_base.py

```diff
@@ -766,6 +766,48 @@ def test_sub(self):
         self.assertRaises(TypeError, lambda: idx - idx.tolist())
         self.assertRaises(TypeError, lambda: idx.tolist() - idx)
 
+    def test_map_identity_mapping(self):
+        # GH 12766
+        for name, cur_index in self.indices.items():
+            self.assert_index_equal(cur_index, cur_index.map(lambda x: x))
```

Comment:
Sounds good.  I changed the ones related to my commit but left the other ones in test_base.py alone.